x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -366,25 +367,31 @@ async function endpointListMiddleware({
       payload: false,
     });
 
-    try {
-      const policyDataResponse: GetPolicyListResponse =
-        await sendGetEndpointSpecificPackagePolicies(http, {
-          query: {
-            perPage: 50, // Since this is an onboarding flow, we'll cap at 50 policies.
-            page: 1,
+    if (canFetchPackageAndAgentPolicies(coreStart.application.capabilities)) {
+      try {
+        const policyDataResponse: GetPolicyListResponse =
+          await sendGetEndpointSpecificPackagePolicies(http, {
+            query: {
+              perPage: 50, // Since this is an onboarding flow, we'll cap at 50 policies.
+              page: 1,
+            },
+          });
+
+        dispatch({
+          type: 'serverReturnedPoliciesForOnboarding',
+          payload: {
+            policyItems: policyDataResponse.items,
           },
         });
-
-      dispatch({
-        type: 'serverReturnedPoliciesForOnboarding',
-        payload: {
-          policyItems: policyDataResponse.items,
-        },
-      });
-    } catch (error) {
+      } catch (error) {
+        dispatch({
+          type: 'serverFailedToReturnPoliciesForOnboarding',
+          payload: error.body ?? error,
+        });
+      }
+    } else {
```

Comment:
Just thinking through this.... I initially thought that the fix should be to add our `canReadEndpointList` to the Package Policy Bulk get API in fleet - just like we do for other privileges. See here: 

https://github.com/elastic/kibana/blob/15c1ceb47539f1a8223c049a26abba7373fd8a10/x-pack/plugins/fleet/server/services/security/route_required_authz.ts#L72-L100

But now, knowing that this call to the API is only used for onboarding, maybe its ok to go with your implementation 🤔

Reply:
i think it wouldn't really make much sense: we try to call this API when there are no endpoints, assuming that there may be already created Defend package policies that could be used to enroll agents to. but, without fleet privileges, users cannot enroll agents, so there's no point in trying to fetch the package policies.

Reply:
Got it. thanks for response. I'm good keeping it as you have it